### ISSUE TITLE
Service won't fully activate if you use the use hello-world image

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rancher#1546]
v0.27.0-rc4
Steps:
1.  Create a service, scale 1 using the hello-world image
2.  Start the service

Results:
The service sits in activating state and never goes to Active.

Expected:
Service should fully activate

**Comment by alena1108** [comment]
hello-world image doesn't have any long running process, and exists run after printing out its "hello world" message. Therefore container goes to stopped state right away.

**Comment by vincent99** [comment]
Yes that's the point; shouldn't we do something about that situation instead of leaving the service in a state that sounds like it's transitioning but will never complete?

It seems impossible to use services for anything that is expected to terminate right now, like volume or image cleanup scripts that you run every once in a while and want scheduled globally.

**Comment by alena1108** [comment]
@vincent99 I've just realized the PR for "Don't auto restart svc containers with restartPolicy=No" would provide the fix (https://github.com/rancher/cattle/pull/730, https://github.com/rancher/rancher/issues/1596). So if  you know that process inside your container terminates shortly, specify restartPolicy=No, and service reconcile process won't attempt to restart it. Tested against hello-world image, works fine. So reopening the ticket; will reassign to-test after the PR gets in.

Support for data only containers also covers a similar use case, but here even the initial container start is not necessary. Here are the corresponding tickets:

https://github.com/rancher/rancher/issues/1378
https://github.com/rancher/rancher/issues/1395
